src/telemetry/metrics/tls_config_reload.rs

```diff
@@ -14,7 +14,10 @@ metrics! {
 }
 
 #[derive(Debug, Default)]
-pub struct TlsConfigReload {
+pub struct TlsConfigReload(RwLock<Inner>);
```

Comment:
TIOLI: It seems a little weird to me to put the `RwLock` inside this struct and the `Arc` outside of it, when seeing the type `Arc<TlsConfigReload>` earlier in this review, I thought, "huh, it's immutable?". This is fairly minor, but it seems to me that either both the sharing and mutability be encapsulated inside this type (and we just pass around an opaque `TlsConfigReload`), or move both of them outside the type (and pass around an `Arc<RwLock<TlsConfigReload>>`). I'm in favour of the first personally.

Reply:
The intent was that that we shouldn't dictate how the users of this hold the reference. If they want to hold a weak reference, let them.  Nor should the lock be exposed to users, as it's very much an implementation detail of how metrics are recorded.  The `Arc` explicitly says "this thing is shard", which is what I'm trying to convey.

That said, I think I may want to move the Arc for another reason.

Reply:
So, is it by design that the public interface of this type can mutate itself through an immutable reference?

Reply:
Basically, yes. I'm going to split this into two halves so the contract is clearer. The point is that this thing could be backed by an mpsc or a rwlock or some other shared datastructure that the writer can't assume exclusive ownership of.